ical.go

```diff
@@ -118,38 +118,34 @@ func (prop *Prop) Bool() (bool, error) {
 
 // DateTime parses the property value as a date-time or a date.
 func (prop *Prop) DateTime(loc *time.Location) (time.Time, error) {
-	// Use the TZID location, if available, otherwise the given location.
-	// Default to UTC, if there is no TZID or given location.
-	if tzid := prop.Params.Get(PropTimezoneID); tzid != "" {
-		tzLoc, err := time.LoadLocation(tzid)
-		if err != nil {
-			return time.Time{}, err
-		}
-		loc = tzLoc
-	}
+	// Default to UTC, if there is no given location.
 	if loc == nil {
 		loc = time.UTC
 	}
 
 	valueType := prop.ValueType()
 	valueLength := len(prop.Value)
-	switch valueType {
-	case ValueDate:
-		return time.ParseInLocation(dateFormat, prop.Value, loc)
-	case ValueDateTime:
-		if valueLength == len(datetimeFormat) {
-			return time.ParseInLocation(datetimeFormat, prop.Value, loc)
-		}
-		return time.ParseInLocation(datetimeUTCFormat, prop.Value, time.UTC)
-	case ValueDefault:
-		switch valueLength {
-		case len(dateFormat):
+	switch valueLength {
```

Comment:
Hm, I'm not a fan of always using the value length here. I'd prefer to rely on `valueType` instead. How about we re-organize this logic a bit, for instance:

```go
if valueType == ValueDefault {
    switch valueLength {
    case len(dateFormat):
        valueType = ValueDate
    case len(datetimeFormat), len(datetimeUTCFormat):
        valueType = ValueDateTime
    }
}
switch valueType {
case ValueDate:
    // same as old logic
case ValueDateTime:
    // same as old logic
}
```

Reply:
Good point. I addressed this in the latest version.